packages/caravan-fees/src/cpfp.ts

```diff
@@ -0,0 +1,216 @@
+import { BtcTransactionTemplate } from "./btcTransactionTemplate";
+import {
+  BtcTxInputTemplate,
+  BtcTxOutputTemplate,
+} from "./btcTransactionComponents";
+import { Network } from "@caravan/bitcoin";
+import { TxOutputType, UTXO, FeeBumpStrategy, Satoshis } from "./types";
+import BigNumber from "bignumber.js";
+
+/**
+ * Creates a Child-Pays-for-Parent (CPFP) transaction to accelerate the confirmation
+ * of an unconfirmed parent transaction.
+ *
+ * This function creates a new transaction (child) that spends an output from the
+ * original unconfirmed transaction (parent). The child transaction includes a higher
+ * fee to incentivize miners to confirm both the parent and child transactions together.
+ *
+ * @param {string} originalTx - The hex-encoded original (parent) transaction to be accelerated.
+ * @param {UTXO[]} availableInputs - Array of available UTXOs, including the spendable output from the parent transaction.
+ * @param {number} spendableOutputIndex - The index of the output in the parent transaction that will be spent in the child transaction.
+ * @param {string} changeAddress - The address where any excess funds (change) will be sent in the child transaction.
+ * @param {Network} network - The Bitcoin network being used (e.g., mainnet, testnet).
+ * @param {Satoshis} dustThreshold - The dust threshold in satoshis. Outputs below this value are considered "dust".
+ * @param {string} scriptType - The type of script used for the transaction (e.g., P2PKH, P2SH, P2WSH).
+ * @param {Satoshis} absoluteFee - The absolute fee of the original transaction in satoshis.
+ * @param {number} requiredSigners - The number of required signers for the multisig setup.
+ * @param {number} totalSigners - The total number of signers in the multisig setup.
+ * @param {boolean} [strict=false] - If true, enforces stricter validation rules.
+ * @returns {string} The base64-encoded PSBT of the CPFP (child) transaction.
+ * @throws {Error} If CPFP is not possible or if the transaction creation fails.
+ *
+ * @example
+ * const cpfpTx = createCPFPTransaction(
+ *   originalTxHex,
+ *   availableUTXOs,
+ *   1, // spendable output index
+ *   'bc1q...', // change address
+ *   Network.MAINNET,
+ *   '546', // dust threshold
+ *   'P2WSH',
+ *   '1000', // absolute fee
+ *   2, // required signers
+ *   3,  // total signers
+ *   true // strict mode
+ * );
+ *
+ * @see https://bitcoinops.org/en/topics/cpfp/ Bitcoin Optech on Child Pays for Parent
+ * @see https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki#implementation-details-1 BIP 125: Implementation details related to CPFP
+ */
+export const createCPFPTransaction = (
+  originalTx: string,
+  availableInputs: UTXO[],
+  spendableOutputIndex: number,
+  changeAddress: string,
+  network: Network,
+  dustThreshold: Satoshis,
+  scriptType: string,
+  absoluteFee: Satoshis,
+  requiredSigners: number,
+  totalSigners: number,
+  strict: boolean = false,
```

Comment:
arity here is not great. Make this one big options object. If you have one argument out of order things will break.

Reply:
Thanks ... changes [done](https://github.com/caravan-bitcoin/caravan/pull/114/commits/a215dea59e6ac918e960757d5dd6f74ed09ae55d#diff-852fd83c241ed78594dc29389ec04d1425d784e665c706434a8de046f7daa85dR56)

and added detailed type for them [too](https://github.com/caravan-bitcoin/caravan/pull/114/commits/345d6fb5310907742dcd172a479d9547eb2fe9e1#diff-593f8d602f801c56261ed7a8dc941947102f6a7b26c483c5ee8cf9c293fffb33R382)